### PR TITLE
Add an iterateLike matcher

### PR DESCRIPTION
**Pull request by sroze**
So we can have non-strict comparison for iterators.

### CODE REVIEW

**Comment by ciaranmcnulty** [comment]
Looks useful - would benefit from some tests + doc changes

**Comment by ciaranmcnulty** [comment]
Also, see #1032

**Comment by sroze** [comment]
@ciaranmcnulty anything special you'd like me to have a look to in this issue? 😃

**Comment by ciaranmcnulty** [comment]
Just linking so I remember to close it :-)

Could you also add an equivalent of the shouldIterateStarting?

**Comment by sroze** [comment]
`shouldIterateStartingLike` ?

File: src/PhpSpec/Matcher/Iterate/IterablesMatcher.php [hunk 1]
```diff
@@ -22,13 +22,14 @@ public function __construct(Presenter $presenter)
...
      * @param array|\Traversable $expected
+     * @param bool               $strict
      *
      * @throws \InvalidArgumentException
      * @throws SubjectElementDoesNotMatchException
      * @throws SubjectHasFewerElementsException
      * @throws SubjectHasMoreElementsException
      */
-    public function match($subject, $expected)
+    public function match($subject, $expected, bool $strict = false)
```
Comment (ciaranmcnulty) [inline src/PhpSpec/Matcher/Iterate/IterablesMatcher.php]: The old behaviour was to be strict, so this should default to `true`

**Comment by ciaranmcnulty** [comment]
Thanks @sroze !

**Comment by sroze** [comment]
Thank you for taking it over @ciaranmcnulty!

**Pull request merged**